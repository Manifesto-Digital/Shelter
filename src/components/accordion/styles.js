import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
