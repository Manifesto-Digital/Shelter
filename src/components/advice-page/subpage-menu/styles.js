import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
