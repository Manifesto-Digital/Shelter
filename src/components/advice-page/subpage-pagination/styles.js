import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
