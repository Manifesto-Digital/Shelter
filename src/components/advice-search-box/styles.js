import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
