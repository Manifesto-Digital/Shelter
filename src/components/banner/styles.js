import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
