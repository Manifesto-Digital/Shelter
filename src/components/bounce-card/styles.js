import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
