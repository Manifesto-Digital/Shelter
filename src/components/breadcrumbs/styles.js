import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
