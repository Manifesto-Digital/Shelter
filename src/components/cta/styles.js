import styled from 'styled-components'

export const ButtonLink = styled.a`
  display: inline-block;
  padding: 0.65em 2em;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'blue' && props.theme.palette.link) ||
    (props.bg === 'white outline' && 'transparent')};

  border: ${props =>
    props.bg === 'white outline'
      ? `1px solid ${props.theme.palette.white}`
      : 'none'};

  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'white outline' && props.theme.palette.white)};
`
