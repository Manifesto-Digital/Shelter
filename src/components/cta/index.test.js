import React from 'react';
import { shallow } from 'enzyme';
import {
  snapshotComponent,
  mountWithTheme,
} from '../../../__tests__/helpers/index';
import CTA from './index';
import { StyledLinkHandler } from './styles';
import {
  createFactory,
  createInternalLink,
  createImage,
} from '../../utils/test-factories';
import { hidePascalCaseWarning } from '../../utils/test-mocks';

// Default props
export const createCTA = createFactory({
  buttonText: 'Button text',
  ctaColour: 'Red',
  internalLink: createInternalLink(),
  externalLink: {
    URL: 'http://example.com',
    newTab: true,
  },
});

it('renders correctly', () => {
  const mockData = createCTA();

  snapshotComponent(<CTA {...CTA.fromCMS(mockData)} />);
});

hidePascalCaseWarning();

it('displays the correct text', () => {
  const mockData = createCTA({ buttonText: 'Mock button text' });
  const wrapper = shallow(<CTA {...CTA.fromCMS(mockData)} />);
  expect(wrapper.find(StyledLinkHandler).text()).toBe(mockData.buttonText);
});

it('displays an icon if provided', () => {
  const mockData = createCTA({ icon: createImage() });
  const wrapper = mountWithTheme(<CTA {...CTA.fromCMS(mockData)} />);
  expect(wrapper.find('img').prop('src')).toContain(mockData.icon.file.url);
});

it('displays width 100% if fullWidth is specified', () => {
  const mockData = createCTA();
  const wrapper = mountWithTheme(<CTA {...CTA.fromCMS(mockData)} fullWidth />);

  expect(wrapper).toHaveStyleRule('width', '100%');
});
