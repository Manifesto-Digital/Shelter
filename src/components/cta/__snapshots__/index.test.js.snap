// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  display: inline-block;
}

.c0 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #ea232d;
  border: none;
  color: #fff;
}

<Link
  className="c0"
  to="/internal-link-slug"
>
  <span
    className="c1"
  >
    Button text
  </span>
</Link>
`;
